pkg/controller/servicebinding/binder.go

```diff
@@ -152,55 +166,68 @@ func (b *binder) removeSpecVolumes(
 		return nil, err
 	}
 	volumes = b.removeVolumes(volumes)
-	if err = unstructured.SetNestedSlice(obj.Object, volumes, b.getVolumesPath()...); err != nil {
+
+	volumesUnstructured, err := runtime.DefaultUnstructuredConverter.ToUnstructured(volumes)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = unstructured.SetNestedMap(obj.Object, volumesUnstructured, b.getVolumesPath()...); err != nil {
```

Comment:
Is the conversion from `volumes` to `volumesUnstructured` required? Perhaps using `unstructured.SetNestedField` with `volumes` as an argument would be simpler.